Added macros for ESP -  WifiGMode and WifiPower to reduce/eliminate Wifi interference on sensors like HC-SR501

diepeterpan opened the pull request:
## Description:

Added macros for ESP -  WifiGMode and WifiPower to reduce/eliminate Wifi interference on sensors like HC-SR501 and updated documentation.

https://community.openmqttgateway.com/t/wifi-interference-with-hc-sr501/2303

## Checklist:
  - [ X] The pull request is done against the latest development branch
  - [ X] Only one feature/fix was added per PR and the code change compiles without warnings
  - [ X] I accept the [DCO](https://github.com/1technophile/OpenMQTTGateway/blob/development/docs/participate/development.md#developer-certificate-of-origin).

diepeterpan committed:
My devices

diepeterpan committed:
Merge branch 'development' of https://github.com/diepeterpan/OpenMQTTGateway into development

diepeterpan committed:
My tuning

agreylin committed:
Fix WiFi interference on PIR

diepeterpan committed:
Enhance custom WiFi setting with defines

diepeterpan committed:
Fixes for lower WiFi signal

diepeterpan committed:
This seems to work

diepeterpan committed:
Merge branch '1technophile:development' into development

diepeterpan committed:
Added to DOCS re: WiFi interference.

diepeterpan committed:
Merge branch 'development' of https://github.com/diepeterpan/OpenMQTTGateway into development

diepeterpan committed:
Try and fix LINT issues ;-)

diepeterpan committed:
Me and lint ;-(

diepeterpan committed:
Getting there with lint

diepeterpan committed:
Oops, need to be on default libs

1technophile commented:
Thanks for the PR, could you rebase and force push please

diepeterpan commented:
> Thanks for the PR, could you rebase and force push please

Apology for the dumb question, Rebase to?
![image](https://user-images.githubusercontent.com/15066052/225118620-72a87ddd-34d2-4430-b90e-58bf8ce98241.png)

diepeterpan committed:
Remove some LAN ip specific and signal cust

diepeterpan commented:
Also don't see a [Update branch] here.
![image](https://user-images.githubusercontent.com/15066052/225125579-9a3aa555-8e9f-486b-b6d3-af7514cc1259.png)

1technophile commented:
Oh, I apologize for the confusion! I thought the added environments were a result of a missing rebase. Just to share my perspective, I would rather not add production environments for specific sensor configurations as it may lead to a lot of unnecessary environments. However, I do agree that we still need to test any modifications that are made to ensure proper build functionality. 
As a solution, how about we keep one environment and add the suffix "-test" to differentiate it from the production environments and prevent it from adding a binary?

1technophile commented on main/main.ino:
```diff
@@ -970,6 +970,27 @@ bool wifi_reconnect_bypass() {
+    }
+#    endif
+
+#    ifdef WifiPower
```
Could you add these macros into User_config.h (commented) with an explanation

diepeterpan replied:
Done.

1technophile commented on main/main.ino:
```diff
@@ -970,6 +970,27 @@ bool wifi_reconnect_bypass() {
+
+    //Reduce WiFi interference when using ESP32 using custom WiFi mode and tx power
+#  ifdef ESP32
+    // https://github.com/jnogues/ESP32-Long-Range-WiFi/issues/4
+    // Reset wifi protocol
+    if (esp_wifi_set_protocol(WIFI_IF_STA, WIFI_PROTOCOL_11B | WIFI_PROTOCOL_11G | WIFI_PROTOCOL_11N | WIFI_PROTOCOL_LR) != ESP_OK) {
```
The goal of this is to improve wifi reconnection ?

diepeterpan replied:
As I understand it when we start changing the Wifi band it gets stored in the ESP32 firmware and needs to be reset to default using explicit code.  Hence the research and reference to https://github.com/jnogues/ESP32-Long-Range-WiFi/issues/4 where other Platform.IO users had the same experience.

I could not understand why my ESP32 would not want to connect to Band N after lowering it to Band G even after removing my code. And this fixed it.

1technophile replied:
We will need testing of this with different router configurations, as I'm not sure of the impact of this function.

1technophile replied:
I just tested it, and it prevents the wifi configuration portal from starting

diepeterpan replied:
Busy testing a solution.

1technophile replied:
If we could avoid going through this function for regular users we would save a bunch of regression testing and potential headache

diepeterpan replied:
I got rid of that; from my testing, it should be working, incl. WiFi configuration.  I did an {"cmd":"erase"} and checked that I can connect and reconfigure.

diepeterpan replied:
See whether the latest changes are more suitable; I have done quite a bit of testing and it seems fine.

1technophile replied:
I'm not a fan of impacting the wifi connection for everybody. We had users struggling with their routers the last time we did that.
Can these changes be only for HC-SR501?

diepeterpan replied:
The latest pre-processor statements prevent this code from impacting existing users.  Only when they **explicitly**  **DEFINE** or **uncomment** either **WifiGMode** or **WifiPower** and have an **ESP32** will the code be compiled which changes the WiFi protocol and transmit power.

1technophile replied:
Great!

diepeterpan commented:
> Oh, I apologize for the confusion! I thought the added environments were a result of a missing rebase. Just to share my perspective, I would rather not add production environments for specific sensor configurations as it may lead to a lot of unnecessary environments. However, I do agree that we still need to test any modifications that are made to ensure proper build functionality. As a solution, how about we keep one environment and add the suffix "-test" to differentiate it from the production environments and prevent it from adding a binary?

Instead of "-test"; which might sound like we are not sure whether it works rather use the suffix "-custom" for custom working configuration without production binaries.

diepeterpan committed:
Add sufficx -custom to configs without prod binaries

diepeterpan committed:
Add ESP32 Wifi band and tx power to user_config.h

And fix _custom

diepeterpan committed:
Fix environment

1technophile commented:
> Instead of "-test"; which might sound like we are not sure whether it works rather use the suffix "-custom" for custom working configuration without production binaries.

`-test` is the current prefix used by the scripts to differentiate the environments between the testing ones and the production ones
https://github.com/1technophile/OpenMQTTGateway/blob/3ceb19f7beb3531f933d8de250deae6a6d14fd98/scripts/prepare_deploy.sh#L20

Users can take samples from either production or test one to build their custom environment.

Also if you are not doing it already I encourage to use a separate file for custom environment like a [prod_env.ini](https://github.com/1technophile/OpenMQTTGateway/blob/development/prod_env.ini.example), this can host the custom configurations, stay local and will be ignored by git.

diepeterpan committed:
Change -custom to -test

diepeterpan commented:
> > Instead of "-test"; which might sound like we are not sure whether it works rather use the suffix "-custom" for custom working configuration without production binaries.
> 
> `-test` is the current prefix used by the scripts to differentiate the environments between the testing ones and the production ones
> 
> https://github.com/1technophile/OpenMQTTGateway/blob/3ceb19f7beb3531f933d8de250deae6a6d14fd98/scripts/prepare_deploy.sh#L20
> 
> Users can take samples from either production or test one to build their custom environment.
> 
> Also if you are not doing it already I encourage to use a separate file for custom environment like a [prod_env.ini](https://github.com/1technophile/OpenMQTTGateway/blob/development/prod_env.ini.example), this can host the custom configurations, stay local and will be ignored by git.

Kewl, changed to -test ; will def. look into prod_env.ini. ; I was wondering how ones custom environment is maintained.

diepeterpan committed:
Lets get rid of Reset wifi protocol

And let lint out on me

diepeterpan committed:
LINT changes

diepeterpan committed:
Merge branch '1technophile:development' into development

diepeterpan committed:
Further reduce risk on non users of new flags

diepeterpan committed:
Now it is in the users hands

No impact on those not using the new settings.  And those that do have full control.

diepeterpan committed:
Dear LINT
